[Docs] Non-Breaking change of "p4c" to "P4C" (#4620)
p4lang/p4c: docs/README.md

## Compiler Driver

> **P4C** is a compiler driver. The goal is to provide a consistent user interface

Reviewer: ```suggestion
**p4c** is a compiler driver. The goal is to provide a consistent user interface
```

p4c_PYTHON += p4c.custom.cfg
```

> There is an global variable `config` in P4C compiler driver that stores the build steps

Reviewer: ```suggestion
There is an global variable `config` in the `p4c` compiler driver that stores the build steps
```

After adding the new configuration file, rerun `bootstrap.sh`

> For testing purpose, P4C will be installed in the build/ directory when executing `make`.
> User can install `P4C` to other system path by running `make install`

Reviewer: ```suggestion
For testing purposes, `p4c` will be installed in the build/ directory when executing `make`.
Users can install `p4c` to other system path by running `make install`
```
